src/management/api/analytics/logs/log.component.ts

```diff
@@ -16,60 +16,108 @@ import NotificationService from '../../../../services/notification.service';
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-const LogComponent: ng.IComponentOptions = {
-  bindings: {
-    log: '<'
-  },
-  controller: function($state: StateService, NotificationService: NotificationService, Constants: any) {
-    'ngInject';
-    this.Constants = Constants;
-    this.NotificationService = NotificationService;
 
+interface IQueryParam {
+  key: string;
+  value: string;
+}
+
+class LogComponentController {
+  public log: {
+    clientRequest: any
+    proxyRequest: any
+    clientResponse: any
+    proxyResponse: any
+  };
+  public backStateParams: { q: any; size: any; from: any; to: any; page: any };
+
+  constructor(
+    public readonly Constants: any,
+    private readonly $state: StateService,
+    private readonly NotificationService: NotificationService,
+  ) {
+    'ngInject';
     this.backStateParams = {
       from: $state.params.from,
       to: $state.params.to,
       q: $state.params.q,
       page: $state.params.page,
-      size: $state.params.size
+      size: $state.params.size,
     };
+  }
 
-    this.$onInit = () => {
-      this.headersAsList(this.log.clientRequest);
-      this.headersAsList(this.log.proxyRequest);
-      this.headersAsList(this.log.clientResponse);
-      this.headersAsList(this.log.proxyResponse);
-    };
 
-    this.headersAsList = (obj) => {
-      if (obj) {
-        obj.headersAsList = [];
-        for (const k in obj.headers) {
-          if (obj.headers.hasOwnProperty(k)) {
-            for (const v in obj.headers[k]) {
-              if (obj.headers[k].hasOwnProperty(v)) {
-                obj.headersAsList.push([k, obj.headers[k][v]]);
-              }
+  $onInit() {
+    this.headersAsList(this.log.clientRequest);
+    this.log.clientRequest = this.computeAndSetQueryParams(this.log.clientRequest);
+    this.headersAsList(this.log.proxyRequest);
+    this.log.proxyRequest = this.computeAndSetQueryParams(this.log.proxyRequest);
+    this.headersAsList(this.log.clientResponse);
+    this.headersAsList(this.log.proxyResponse);
+  }
+
+  getMimeType(log) {
+    if (log.headers['Content-Type'] !== undefined) {
```

Comment:
I suggest this:
```suggestion
    if (log.headers['Content-Type'] != undefined) {
```

if `log.headers['Content-Type']` is null, you will enter if scope and try to access index 0 of this array.

wdyt ?

Reply:
I improved it a bit more by checking `log.headers['Content-Type']` is an array.